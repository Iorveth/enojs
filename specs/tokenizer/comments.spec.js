const tokenize = require('../../lib/parse_steps/tokenize.js');

const input = `
> note
> more notes
    >    note
    >
`.trim();

describe('Comment tokenization', () => {
  it('performs to specification', () => {
    const context = { input: input };

    tokenize(context);

    expect(context.instructions).toMatchSnapshot();
  });
});
