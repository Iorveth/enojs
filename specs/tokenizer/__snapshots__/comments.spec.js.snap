// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comment tokenization performs to specification 1`] = `
Array [
  Object {
    "comment": "note",
    "index": 0,
    "length": 6,
    "line": 0,
    "ranges": Object {
      "comment": Array [
        2,
        6,
      ],
      "commentOperator": Array [
        0,
        1,
      ],
    },
    "type": "NOOP",
  },
  Object {
    "comment": "more notes",
    "index": 7,
    "length": 12,
    "line": 1,
    "ranges": Object {
      "comment": Array [
        2,
        12,
      ],
      "commentOperator": Array [
        0,
        1,
      ],
    },
    "type": "NOOP",
  },
  Object {
    "comment": "note",
    "index": 20,
    "length": 13,
    "line": 2,
    "ranges": Object {
      "comment": Array [
        9,
        13,
      ],
      "commentOperator": Array [
        4,
        5,
      ],
    },
    "type": "NOOP",
  },
  Object {
    "comment": null,
    "index": 34,
    "length": 5,
    "line": 3,
    "ranges": Object {
      "commentOperator": Array [
        4,
        5,
      ],
    },
    "type": "NOOP",
  },
]
`;
